src/java.base/share/classes/java/text/DecimalFormat.java

```diff
@@ -55,45 +55,65 @@
+ * a decimal format. For example, {@code "2,000.5"}.
+ * <li> Use {@link #getIntegerInstance()} to get an integer number format.
+ * For example, {@code "2,000"}.
+ * <li> Use {@link #getCurrencyInstance} to get a currency number format.
+ * For example, {@code "$2,000.50"}.
+ * <li> Use {@link #getPercentInstance} to get a format for displaying percentages.
+ * For example, {@code "200,050%"}.
+ * <li> Use one of the {@code DecimalFormat} constructors to obtain a {@code
+ * DecimalFormat} with further customization.
+ * </ul>
  *
- * <h2 id="patterns">Patterns</h2>
+ * <small>Note: It is recommended to use one of the NumberFormat factory methods
+ * which is tailored to the conventions of the given locale to retrieve a
+ * DecimalFormat.</small>
+ *<p>
+ * Alternatively, if a {@code DecimalFormat} for a different locale is required, use
+ * one of the factory method variants that take {@code locale} as a parameter,
+ * for example, {@link #getIntegerInstance(Locale)}. {@link
+ * NumberFormat#getAvailableLocales()} can be used to determine if the desired
+ * locale is supported.
  *
- * Note: For any given {@code DecimalFormat} pattern, if the pattern is not
- * in scientific notation, the maximum number of integer digits will not be
- * derived from the pattern, and instead set to {@link Integer#MAX_VALUE}.
- * Otherwise, if the pattern is in scientific notation, the maximum number of
- * integer digits will be derived from the pattern. This derivation is detailed
- * in the {@link ##scientific_notation Scientific Notation} section. This behavior
- * is the typical end-user desire; {@link #setMaximumIntegerDigits(int)} can be
- * used to manually adjust the maximum integer digits.
+ * <h2>Using DecimalFormat</h2>
+ * The following is an example of formatting and parsing in a customized localized fashion,
+ *
+ * {@snippet lang=java :
+ * DecimalFormat currencyFormat = (DecimalFormat) NumberFormat.getCurrencyInstance(Locale.US);
+ * // The positive prefix is implicitly "$" for a US currency instance
+ * currencyFormat.setPositiveSuffix(" dollars");
+ * currencyFormat.format(100000); // returns "$100,000.00 dollars"
+ * currencyFormat.parse("$100,000.00 dollars"); // returns 100000
+ * }
+ *
+ * <h2 id="patterns">DecimalFormat Pattern</h2>
+ *
+ * A {@code DecimalFormat} comprises a <em>pattern</em> and a set of
+ * <em>symbols</em>. The pattern may be set directly using {@code applyPattern()},
+ * or indirectly using the various API methods. The symbols are stored in a {@code
+ * DecimalFormatSymbols} object. When using the {@code NumberFormat} factory
+ * methods, the pattern and symbols are read from localized {@code ResourceBundle}s,
+ * adhering to the given locale's conventions.
+ * <p><b>For those planning to only use the factory methods, the pattern syntax may
+ * not be relevant. If this is the case, continue reading at the {@link ##formatting Formatting and Parsing}
+ * section.</b>
```

Comment:
As in the `CompactNumberFormat`, I suggest moving the `DecimalFormat Pattern` section at the end.